_local/Dockerfile

```diff
@@ -0,0 +1,38 @@
+ARG GOTAG
+
+FROM golang:${GOTAG}
+
+ARG LOCUSER
+ARG UID
+ARG GID
+
+RUN apt update && \
+    apt install -y autoconf \
+        gettext \
+        libcrypt-openssl* \
+        libcurl4-open* \
+        libexpat1* \
+        libzlcore-* \
+        libzltext-* \
+        tcl8.6* \
+        zlib1g-dev
+
+# worktree_test.go:677:
+# c.Assert(idx.Entries[0].UID, Not(Equals), uint32(0))
+# User MUST NOT == uid=0(root) gid=0(root) groups=0(root)
+RUN DEBIAN_FRONTEND=noninteractive; \
+    addgroup --gid ${GID} --force-badname ${LOCUSER}; \
+    adduser --uid ${UID} --ingroup ${LOCUSER} \
+        --disabled-password --quiet --force-badname ${LOCUSER}
```

Comment:
Instead of having to create an user, we can just run it as nobody:

```
USER 65534:65534
```

Reply:
@pjbgf It looks like the `git config` will fail, I tested with your suggestion ...

```shell
GO_VERSION=1.19 gmake test-go
export BASH_ENV=/Users/patrick.miller/src/github.com/patgmiller/go-git/.git-dist/.env
/usr/bin/env bash _local/test-go.sh 1.19
Running checks for: darwin
[+] Building 0.2s (8/9)                                                                                                                                            docker:desktop-linux
 => [internal] load .dockerignore                                                                                                                                                  0.0s
 => => transferring context: 57B                                                                                                                                                   0.0s
 => [internal] load build definition from Dockerfile                                                                                                                               0.0s
 => => transferring dockerfile: 963B                                                                                                                                               0.0s
 => [internal] load metadata for docker.io/library/golang:1.19-bullseye                                                                                                            0.0s
 => [1/5] FROM docker.io/library/golang:1.19-bullseye                                                                                                                              0.0s
 => [internal] load build context                                                                                                                                                  0.0s
 => => transferring context: 156B                                                                                                                                                  0.0s
 => CACHED [2/5] RUN apt update &&     apt install -y autoconf         gettext         libcrypt-openssl*         libcurl4-open*         libexpat1*         libzlcore-*         li  0.0s
 => [3/5] COPY --chown=root:root _local/* /                                                                                                                                        0.0s
 => ERROR [4/5] RUN git config --global user.email "gha@example.com";     git config --global user.name "GitHub Actions"                                                           0.1s
------
 > [4/5] RUN git config --global user.email "gha@example.com";     git config --global user.name "GitHub Actions":
0.127 error: could not lock config file /nonexistent/.gitconfig: No such file or directory
0.128 error: could not lock config file /nonexistent/.gitconfig: No such file or directory
------
Dockerfile:33
--------------------
  32 |
  33 | >>> RUN git config --global user.email "gha@example.com"; \
  34 | >>>     git config --global user.name "GitHub Actions"
  35 |
--------------------
ERROR: failed to solve: process "/bin/sh -c git config --global user.email \"gha@example.com\";     git config --global user.name \"GitHub Actions\"" did not complete successfully: exit code: 255
gmake: *** [Makefile:62: test-go] Error 1
```

 and there by the test i added it for as well as the `ssh-keyscan`, not sure this later part is an issue but i recall the user config being related to a number of tests. 

```shell
# worktree_test.go:677:
# c.Assert(idx.Entries[0].UID, Not(Equals), uint32(0))
# User MUST NOT == uid=0(root) gid=0(root) groups=0(root)
```

There was another reason I added the user, it was because some `.tmp` are created during tests and if the owner id doesn't match the `clean` has to be run with `sudo`.  

Any other thoughts or alternative ideas?